### ISSUE TITLE
Erroneous outputs for scipy.special.mathieu_cem and scipy.special.mathieu_sem

### ISSUE THREAD

**Issue by cw-tan** [scipy/scipy#14543]
Hi, this is to report the inconsistent behavior of the Mathieu functions for larger values of q. For some large fixed 'q's, it is observed that the Mathieu function given by scipy.special.mathieu_cem and scipy.special.mathieu_sem is zero everywhere for certain orders 'n'.

For example, consider the case of q=1200, the following plots are generated for n=51 to n=55.
```
import numpy as np
import matplotlib.pyplot as plt
from scipy.special import mathieu_cem, mathieu_sem

q = 1200

fig, axs = plt.subplots(figsize=(15,3), ncols = 5, nrows = 2, sharey=True, gridspec_kw={'wspace': 0, 'hspace': 0})
for (i,n) in [(0,51), (1,52), (2,53), (3,54), (4,55)]:
  x = np.linspace(-np.pi, np.pi, 1000)
  axs[0][i].plot(x, mathieu_sem(n, q, x*180/np.pi)[0])
  axs[0][i].set_title('n={}'.format(n))
axs[0][0].set_ylabel('mathieu_sem')

for (i,n) in [(0,51), (1,52), (2,53), (3,54), (4,55)]:
  x = np.linspace(-np.pi, np.pi, 1000)
  axs[1][i].plot(x, mathieu_cem(n, q, x*180/np.pi)[0])
axs[1][0].set_ylabel('mathieu_cem')
plt.show()
```
![mathieu](https://user-images.githubusercontent.com/87742566/128228806-1485466d-5eed-460c-b972-cf3d6f6f0e7f.png)

At first, I thought that this was the result of some upper limit in the 'q's and 'n's for which SciPy can output Mathieu functions, but the fact that it works for n=51, 53 and 54, but not n=52 indicates some other cause, I think. 

(I'm using Python 3.7.7, numpy-1.21.1 and scipy-1.7.1)

**Comment by czgdp1807** [comment]
I am able to reproduce the above results in my system (`master` branch). I will try to add the fix in https://github.com/scipy/scipy/pull/14577
Let me know if that works.

**Comment by cw-tan** [comment]
That would be great, thanks!

Looking at the wrapper `mathieu_cem`
https://github.com/scipy/scipy/blob/f57e5bf577c0d2ac6c611047961269348fd49315/scipy/special/functions.json#L943

and subsequently

https://github.com/scipy/scipy/blob/f57e5bf577c0d2ac6c611047961269348fd49315/scipy/special/specfun_wrappers.c#L455

there's a call to 

https://github.com/scipy/scipy/blob/f57e5bf577c0d2ac6c611047961269348fd49315/scipy/special/specfun_wrappers.c#L483


whose Fortran code is

https://github.com/scipy/scipy/blob/42e50b42e3bccabf4e8ba9c409ea3a12dfe4a0ac/scipy/special/specfun/specfun.f#L6528

Seeing as there is a dependence on 

https://github.com/scipy/scipy/blob/42e50b42e3bccabf4e8ba9c409ea3a12dfe4a0ac/scipy/special/specfun/specfun.f#L6542

it is possible that the problem is related to the issue that #14577 is fixing.

**Comment by czgdp1807** [comment]
Yeah. The problem is with iterative methods here. I am looking into the problem.

**Comment by j-bowhay** [comment]
<img width="1252" height="289" alt="Image" src="https://github.com/user-attachments/assets/de7f0838-8f78-405c-a985-ced4b94a4fe5" />
Fix on main!
